platform/mellanox/get_component_versions/get_component_versions.py

```diff
@@ -91,33 +103,50 @@ def get_platform_component_versions():
 
 def get_current_version(comp):
     version = ""
-    # If there's only one command
-    if len(COMMANDS_FOR_ACTUAL[comp]) == 2:
-        version = subprocess.run(COMMANDS_FOR_ACTUAL[comp][0], shell=False, stdout=subprocess.PIPE, text=True)
-        version = str(version.stdout)
-    #If there are two commands and we need a pipe
-    elif len(COMMANDS_FOR_ACTUAL[comp]) == 3:
-        version = check_output_pipe(COMMANDS_FOR_ACTUAL[comp][0], COMMANDS_FOR_ACTUAL[comp][1])
-    parsed_version = re.search(COMMANDS_FOR_ACTUAL[comp][-1], version)
-    return parsed_version.group(1) if parsed_version else "N/A"
+    try:
+        # If there's only one command
+        if len(COMMANDS_FOR_ACTUAL[comp]) == 2:
+            version = subprocess.run(COMMANDS_FOR_ACTUAL[comp][0], shell=False, stdout=subprocess.PIPE, text=True)
+            version = str(version.stdout)
+        #If there are two commands and we need a pipe
+        elif len(COMMANDS_FOR_ACTUAL[comp]) == 3:
+            version = check_output_pipe(COMMANDS_FOR_ACTUAL[comp][0], COMMANDS_FOR_ACTUAL[comp][1])
+        parsed_version = re.search(COMMANDS_FOR_ACTUAL[comp][-1], version)
+        return parsed_version.group(1) if parsed_version else "N/A"
+    except Exception:
+        return "N/A"
 
 
 def format_output_table(table):
     return tabulate(table, HEADERS)
 
 
+def is_simx_switch():
+    simx_version = get_current_version("SIMX")
+
+    return simx_version != 'N/A'
+
+
 def main():
 
     if os.getuid() != 0:
         print("Error: Root privileges are required")
         return
 
     compiled_versions = parse_compiled_components_file()
-    platform_versions = get_platform_component_versions()
+
+    if is_simx_switch():
+        platform_versions = UNAVAILABLE_PLATFORM_VERSIONS
+    else:  
+        platform_versions = get_platform_component_versions()
 
     output_table = []
     for comp in compiled_versions.keys():
         actual = get_current_version(comp)
+
+        if comp == "SIMX" and actual == "N/A":
+            continue
```

Comment:
@skr31 no need this magic, just pull out SimX from the `compiled_versions` dict if platform is matched

Reply:
Fixed